autoscheduler/frontend/src/tests/ui/CourseSelectCard.test.tsx

```diff
@@ -463,4 +569,93 @@ describe('Course Select Card UI', () => {
+      store.dispatch(setTerm('201931'));
+      const {
+        getByLabelText, getByText, findByText, findAllByText,
+      } = render(
+        <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+      );
+
+      // act
+      // fill in course and go to section view
+      const courseEntry = getByLabelText('Course') as HTMLInputElement;
+      fireEvent.change(courseEntry, { target: { value: 'CSCE ' } });
+      fireEvent.click(await findByText('CSCE 121'));
+
+      fireEvent.click(getByText('Section'));
+
+      // assert
+      // verify text for seats is correct
+      const sectionText = (await findAllByText(/ seats left/))[0];
+      expect(sectionText).toHaveTextContent('25/25 seats left');
+    });
+
+    test('when a section has max enrollment', async () => {
+      // arrange
+      fetchMock.mockResponseOnce(JSON.stringify({ // api/course/search
+        results: ['CSCE 121', 'CSCE 221', 'CSCE 312'],
+      }));
+      fetchMock.mockImplementationOnce(testFetch); // api/sections
+
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      store.dispatch(setTerm('201931'));
+      const {
+        getByLabelText, getByText, findByText, findAllByText,
+      } = render(
+        <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+      );
+
+      // act
+      // fill in course and go to section view
+      const courseEntry = getByLabelText('Course') as HTMLInputElement;
+      fireEvent.change(courseEntry, { target: { value: 'CSCE ' } });
+      fireEvent.click(await findByText('CSCE 121'));
+
+      fireEvent.click(getByText('Section'));
+
+      // assert
+      // verify text for seats is correct
+      const sectionText = (await findAllByText(/ seats left/))[1];
+      expect(sectionText).toHaveTextContent('0/25 seats left');
+    });
+
+    test('when a section has over max enrollement', async () => {
+      // arrange
+      fetchMock.mockResponseOnce(JSON.stringify({ // api/course/search
+        results: ['CSCE 121', 'CSCE 221', 'CSCE 312'],
+      }));
+      fetchMock.mockImplementationOnce(testFetch); // api/sections
+
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      store.dispatch(setTerm('201931'));
+      const {
+        getByLabelText, getByText, findByText, findAllByText,
+      } = render(
+        <Provider store={store}><CourseSelectCard id={0} /></Provider>,
+      );
+
+      // act
+      // fill in course and go to section view
+      const courseEntry = getByLabelText('Course') as HTMLInputElement;
+      fireEvent.change(courseEntry, { target: { value: 'CSCE ' } });
+      fireEvent.click(await findByText('CSCE 121'));
+
+      fireEvent.click(getByText('Section'));
+
+      // assert
+      // verify text for seats is correct
+      const sectionText = (await findAllByText(/ seats left/))[2];
+      expect(sectionText).toHaveTextContent('-1/25 seats left');
```

Comment:
Part of the reason I made this test is that I was wondering how we wanted to display sections that have more seats taken than there are total (which I've noticed happens pretty frequently when trying to register for classes in the past), and I can see arguments for either way. Showing negative seats gives more information and is technically correct, while showing "0/25 seats left" instead would look nicer. Feel free to share your thoughts if you think I should change this